server/src/main/java/org/elasticsearch/script/field/WriteField.java

```diff
@@ -58,30 +58,43 @@ public boolean exists() {
      * Move this path to another path in the map.
      *
      * @throws IllegalArgumentException if the other path has contents
-     * @throws UnsupportedOperationException
      */
-    public void move(String path) {
-        throw new UnsupportedOperationException("unimplemented");
+    public WriteField move(String path) {
+        WriteField dest = new WriteField(path, rootSupplier);
+        if (dest.isEmpty() == false) {
+            throw new IllegalArgumentException("Cannot move to non-empty destination [" + path + "]");
+        }
+        return overwrite(path);
     }
 
     /**
      * Move this path to another path in the map, overwriting the destination path if it exists
-     *
-     * @throws UnsupportedOperationException
      */
-    public void overwrite(String path) {
-        throw new UnsupportedOperationException("unimplemented");
+    public WriteField overwrite(String path) {
+        Object value = get(MISSING);
+        remove();
+        setPath(path);
+        if (value == MISSING) {
+            // remove existing mapping if it exists
```

Comment:
nit: should this comment say the opposite?

Reply:
Should we document that the overwrite will remove the path if it's empty?

Reply:
Fixed wording on comment.  Added javadoc about removing the path.